### PR TITLE
Allow admin to check in person from the UI

### PR DESCRIPTION
**Pull request by denys-medynskyi**
Resolves [github issue](https://github.com/pivorakmeetup/pivorak-web-app/issues/684)

### Description

read issue description ☝️ 

- Add `Check in` word and allow admin to `check in` user after event is ended.
- Visit requests will be ordered by `user.first/last name`
- Remove check by `visit request` status while `check in`. Let's make it simpler.

### How to test instructions

- Visit http://localhost:3000/admin/events/pivorak-elixir-edition/visit_requests
- Check in button is on the right in `actions` column 

### Screenshots

screen shot 2018-12-29 at 2 26 44 pm

### CODE REVIEW

File: app/controllers/admin/visit_requests_controller.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def event
     end
 
     def visit_requests
-      @visit_requests ||= search_against(base_scope).includes(:user).order(:id)
+      @visit_requests ||= search_against(base_scope).order('users.first_name, users.last_name').includes(:user).order(:id)
```
Comment (sveredyuk) [inline app/controllers/admin/visit_requests_controller.rb]: why double ordering?

**Review by sveredyuk — COMMENTED** [review]
I think we also need the possibility check-in not attended visitors...

**Review by oookli — APPROVED** [review]

File: app/controllers/admin/visit_request/check_in_controller.rb [hunk 2]
```diff
@@ -5,7 +5,11 @@ class CheckInController < ::Admin::BaseController
 
       def show
```
Comment (floor114) [inline app/controllers/admin/visit_request/check_in_controller.rb:6]: On visit requests page we have real-time updating (actioncable), and in that case, you need to respond correctly. As it is in `ToggleVisitController` and other similar.

**Review by floor114 — CHANGES_REQUESTED** [review]

File: app/controllers/admin/visit_request/check_in_controller.rb (hunk 2 above)
Comment (denys-medynskyi) [reply app/controllers/admin/visit_request/check_in_controller.rb:6]: Totally forgot about it. Thanks 🙇 
Will use `ToggleVisit` endpoint then

File: app/helpers/admin/visit_requests_helper.rb [hunk 3]
```diff
@@ -46,16 +46,15 @@ def visit_request_list_link(visit_request)
...
     end
 
-    def visit_request_visited_link(visit_request)
-      label      = visit_request.visited? ? 'waiting' : 'hello'
-      btn_class  = visit_request.visited? ? 'grey' : 'green'
-      i18n_label = t(label, scope: 'visit_requests')
-
-      link_to i18n_label,
+    def visit_request_check_in_link(visit_request)
+      if !visit_request.checked_in_at
```
Comment (floor114) [inline app/helpers/admin/visit_requests_helper.rb]: I'd add `checked_in?` method to `visit_request`

File: config/environments/development.rb [hunk 4]
```diff
@@ -69,7 +69,7 @@
   # }
 
   config.after_initialize do
-    Bullet.enable = true
```
Comment (floor114) [inline config/environments/development.rb]: revert

File: app/controllers/admin/visit_request/toggle_visit_controller.rb [hunk 5]
```diff
@@ -2,6 +2,7 @@ module Admin
   class VisitRequest
     class ToggleVisitController < VisitRequest::BaseController
       def update
+        ::VisitRequest::CheckIn.new(visit_request).call
```
Comment (floor114) [inline app/controllers/admin/visit_request/toggle_visit_controller.rb]: `::VisitRequest::CheckIn.call(visit_request)`

**Comment by floor114** [comment]
And after moving check-in logic to `ToggleVisitController`, do we need that `CheckInController` and all its logic?

**Comment by denys-medynskyi** [comment]
@floor114 at the moment we need, because when we scan QR code we receive and URL to make a request like this: `GET /check_in/:token`.
And `CheckInController` renders different views depending on the request status.

Maybe, we can merge them, but for now we can go with 2 controllers.

I will think about it or you can take initiative if you want to 😄

**Comment by denys-medynskyi** [comment]
@floor114 take a look again please 🙏

**Review by floor114 — APPROVED** [review]

**Pull request merged**